v2/thirdparty/custom-ui/thirdparty-login.mdx

```diff
@@ -629,127 +633,211 @@ On success, the backend will send back session tokens as part of the response he
+    if (account == null) {
+        print("Google sign in was aborted");
+        return;
+    }
 
-You can use the [react-native-app-auth library](https://github.com/FormidableLabs/react-native-app-auth) in your app to integrate with Google and get the user's access token. The minimum scope that is required by SuperTokens is one that gives you the user's email - in case of Google login, it is `"https://www.googleapis.com/auth/userinfo.email"`.
+    String? authCode = account.serverAuthCode;
 
-:::important
-When setting up on the Google dashboard, you need to generate OAuth credentials for an Android application which will be used for both iOS and Android. Google does not provide a client secret in this case, so on the backend, you can omit the `clientSecret` config in the `clients` array.
-:::
+    if (authCode == null) {
+        print("Google sign in did not return a server auth code");
+        return;
+    }
 
-<h3>Step 2) Calling the signinup API to use the OAuth tokens</h3>
+    // Refer to step 2
+  }
+```
+
+</TabItem>
+
+</FrontendMobileSubTabs>
+
+<h4>Step 2) Calling the signinup API to consume the authorisation token</h4>
 
 <AppInfoForm askForAPIDomain showMultiTenancyTab>
 
-Once you have the `access_token` or the `id_token` from the auth provider, you need to call the signinup API exposed by our backend SDK as shown below:
+Once you have the authorisation code from the auth provider, you need to call the signinup API exposed by our backend SDK as shown below:
 
 ```bash
 curl --location --request POST '^{form_apiDomain}^{form_apiBasePath}/signinup' \
 --header 'rid: ^{codeImportRecipeName}' \
 --header 'Content-Type: application/json; charset=utf-8' \
 --data-raw '{
-    "thirdPartyId": "google",
+    "thirdPartyId": "apple",
     "clientType": "...",
-    "oAuthTokens": {
-        "access_token": "..."
-    },
+    "redirecURIInfo": {
+        "redirectURIOnProviderDashboard": "^{form_apiDomain}^{form_apiBasePath}/callback/apple",
+        "redirectURIQueryParams": {
+            "code": "...",
+            "user": {
+                "name":{
+                    "firstName":"...",
+                    "lastName":"..."
+                },
+                "email":"..."
+            }
+        }
+    }
 }'
 ```
 
 :::important
-- If you don't have the `access_token`, you can send the `id_token` instead.
+- On iOS, the client id set in the backend should be the same as the bundle identifier for your app.
 - The `clientType` input is optional, and is required only if you have initialised more than one client in the provide on the backend (See the "Social / SSO login for both, web and mobile apps" section below).
+- On iOS, `redirectURIOnProviderDashboard` doesn't matter and its value can be a universal link configured for your app.
+- On Android, the `redirectURIOnProviderDashboard` should match the one configured on the Apple developer dashboard.
+- The `user` object contains information provided by Apple.
 :::
```

Comment:
just mention that the redirectURI can be empty and mention why.

Reply:
Say why - it uses the native cllient id to get the auth code, which doesn't require a redirection, so therefore it's not needed